src/library/librarycontrol.cpp

```diff
@@ -545,28 +576,49 @@ void LibraryControl::slotToggleSelectedSidebarItem(double v) {
 }
 
 void LibraryControl::slotGoToItem(double v) {
-    if (!m_pLibraryWidget) {
+    VERIFY_OR_DEBUG_ASSERT(m_pSidebarWidget) {
         return;
     }
-    // Load current track if a LibraryView object has focus
-    const auto activeView = m_pLibraryWidget->getActiveView();
-    if (activeView && activeView->hasFocus()) {
-        return slotLoadSelectedIntoFirstStopped(v);
+    VERIFY_OR_DEBUG_ASSERT(m_pLibraryWidget) {
+        return;
     }
 
-    // Focus the library if this is a leaf node in the tree
-    if (m_pSidebarWidget && m_pSidebarWidget->hasFocus()) {
-        if (v > 0 && m_pSidebarWidget->isLeafNodeSelected()) {
-            setLibraryFocus();
-        } else {
-            // Otherwise toggle the sidebar item expanded state
-            slotToggleSelectedSidebarItem(v);
+    if (v > 0) {
+
+        // Focus the library if this is a leaf node in the tree
+        if (m_pSidebarWidget->hasFocus()) {
+            // ToDo can't expand Tracks and AutoDJ, always returns false for those root items
+            if (m_pSidebarWidget->isLeafNodeSelected()) {
+                return focusLibraryView();
+            } else {
+                // Otherwise toggle the sidebar item expanded state
+                slotToggleSelectedSidebarItem(v);
+            }
         }
-    }
-    // TODO(xxx) instead of remote control the widgets individual, we should 
-    // translate this into Alt+Return and handle it at each library widget 
-    // individual https://bugs.launchpad.net/mixxx/+bug/1758618
+
+        // Load current track if a LibraryView object has focus
+        if (libraryViewHasFocus()) {
+            return slotLoadSelectedIntoFirstStopped(v);
+        }
+
+        // Clear the search if the searchbox has focus
+        if (m_pSearchbox->hasFocus()) {
+            return emit(clearSearch());
+        }
+
+    // TODO(xxx) instead of remote control the widgets individual, we should
+    // translate this into Alt+Return and handle it at each library widget
+    // individually https://bugs.launchpad.net/mixxx/+bug/1758618
     //emitKeyEvent(QKeyEvent{QEvent::KeyPress, Qt::Key_Return, Qt::AltModifier});
+    }
+}
+
+bool LibraryControl::libraryViewHasFocus() {
+    // This always returns false for Recording and Analysis
+    if (m_pLibraryWidget->getActiveView()->hasFocus()) {
+        return true;
+    }
+    return false;
```

Comment:
maybe this doesn't need to be a separate function, until now we need it only twice here.